index/index.js

```diff
@@ -24,85 +24,177 @@ function weekDiff(start, end) {
+    InvocationType: 'RequestResponse',
+    Payload: JSON.stringify(input)
+  };
+  const result = await invokeLambda(params);
+  return result;
+}
+
+// Gets the Campus codes 
+async function getCodes() {
+  let campusParam = {
+    "school": "temple",
+    "term": 202036,
+    "method": "getCampuses",
+    "params": {}
+  };
 
-  // Input for banner proxy lambda 
-  let input = {
+  let params = {
     FunctionName: 'arn:aws:lambda:us-east-2:741865850980:function:banner-proxy:live',
     InvocationType: 'RequestResponse',
-    Payload: JSON.stringify(event)
+    Payload: JSON.stringify(campusParam)
   };
+  const result = invokeCampusLambda(params);
+  return result;
+}
+
+// Formats the class information for the database
+async function formatSections(sections, event) {
+
+    // Iterates through all the sections
+    for (var sectionIndex = 0; sectionIndex < 25; sectionIndex++) {
+      var value = sectionIndex + parseInt(event.params.offset,10);
+      if(value == sections.totalCount) {
+        break;
+      }
 
- // Invokes the banner lambda and gets the payload
-  const result = await invokeLambda(input);
+      // Gets all needed information for all the meeting times of a section
+      var meetingTimes = [];
+      var meetingLength = sections.data[sectionIndex].meetingsFaculty.length;
   
+      for (var meetingIndex = 0; meetingIndex < meetingLength; meetingIndex++) {
+        
+        // Calculates how many weeks for each meeting time
+        var start = sections.data[sectionIndex].meetingsFaculty[meetingIndex].meetingTime.startDate;
+        var end = sections.data[sectionIndex].meetingsFaculty[meetingIndex].meetingTime.endDate;
+        var diff = weekDiff(start, end);
+        
+        // Finds campus code for each meeting time
+        var campus;
 
-  // Loops through the payload (Need to change the for loop values)
-  for (var index = parseInt(event.params.offset); index < parseInt(event.params.pageSize); index++) {
-    if(index == result.totalCount) {
-      break;
-    }
-    var len = result.data[index].meetingsFaculty.length;
-    var times = [];
-
-    for (var index2 = 0; index2 < len; index2++) {
-      // Calculates how many weeks for each course
-      var start = result.data[index].meetingsFaculty[index2].meetingTime.startDate;
-      var end = result.data[index].meetingsFaculty[index2].meetingTime.endDate;
-      var diff = weekDiff(start, end);
-      
-      // Gets faculty information for each course
-      var staff = [];
-      var len2 = result.data[index].faculty.length;
-      for(var j = 0; j < len2; j++) {
-        staff.push(result.data[index].faculty[j].displayName);
+        var campusesLength = campuses.length;
+        for(var index = 0; index < campusesLength; index++) {
+          if(String(sections.data[sectionIndex].campusDescription).localeCompare(String(campuses[index].description)) == 0) {
+            campus = campuses[index].code;
+          }
+       }
```

Comment:
It shouldn't be necessary to reverse-engineer the campus code, there is already a campus code inside the meetingTime objects. (They should be the same for each meetingTime in the section).